### PR TITLE
chore(EMS-2798): No PDF - Your buyer - Move fields to buyerContact and buyerRelationship tables

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This ticket moves some fields from buyer table to buyerContact and buyerRelationship tables

## Resolution :heavy_check_mark:
- Added to schema and sql dump
- Updated ui query and mutation
- Added updated mapAndSaves and save-data
- updated tests

## Miscellaneous :heavy_plus_sign:
- Removed unneeded buyer contact fields from company-or-organisation page

### CODE REVIEW

File: database/exip.sql [hunk 1]
```diff
@@ -215,16 +215,64 @@ CREATE TABLE `Buyer` (
...
+
+
+# Dump of table BuyerRelationship
+# ------------------------------------------------------------
+
+DROP TABLE IF EXISTS `BuyerRelationship`;
+
+CREATE TABLE `BuyerRelationship` (
+  `id` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
+  `exporterIsConnectedWithBuyer` tinyint(1) DEFAULT NULL,
```
Comment (ttbarnes) [inline database/exip.sql:262]: The `Buyer` table still has some of these fields?

File: src/ui/server/controllers/insurance/your-buyer/map-and-save/buyer-relationship/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import hasFormData from '../../../../../helpers/has-form-data';
+import save from '../../save-data/buyer-relationship';
+import { Application, RequestBody, ValidationErrors } from '../../../../../../types';
+import mapSubmittedData from '../../map-submitted-data/buyer-relationship';
+
+/**
+ * maps buyerRelationship request and calls save function
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/map-and-save/buyer-relationship/index.ts:7]: Function name at the top

File: src/ui/server/controllers/insurance/your-buyer/save-data/buyer-relationship/index.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * @returns {Object} saveResponse from API
+ */
+const buyerRelationship = async (application: Application, formBody: RequestBody, errorList?: object) => {
+  // determines which fields to save
+  const dataToSave = stripEmptyFormFields(getDataToSave(formBody, errorList), nullOrEmptyStringFields);
+
+  // sanitise the form data.
+  const sanitisedData = sanitiseData(dataToSave);
+
+  const buyerRelationshipId = application.buyer.buyerRelationship?.id;
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/save-data/buyer-relationship/index.ts:27]: Maybe `application.buyerRelationship` ? Same as above

File: src/ui/templates/insurance/your-buyer/company-or-organisation.njk [hunk 4]
```diff
@@ -148,129 +148,6 @@
...
-        {{ yesNoRadioButtons.render({
-          fieldId: FIELDS.CAN_CONTACT_BUYER.ID,
-          legendText: FIELDS.CAN_CONTACT_BUYER.LABEL,
-          legendClass: 'govuk-fieldset__legend--m',
-          legendSpan: true,
-          hintText: FIELDS.CAN_CONTACT_BUYER.HINT,
-          submittedAnswer: submittedValues[FIELDS.CAN_CONTACT_BUYER.ID] or application.buyer[FIELDS.CAN_CONTACT_BUYER.ID],
-          errorMessage: validationErrors.errorList[FIELDS.CAN_CONTACT_BUYER.ID],
-          dataCyLegend: FIELDS.CAN_CONTACT_BUYER.ID + '-label'
-        }) }}
```
Comment (ttbarnes) [inline src/ui/templates/insurance/your-buyer/company-or-organisation.njk:273]: Looks like some nunjuck imports can be removed now

File: database/exip.sql (hunk 1 above)
Comment (Zainzzkk) [reply database/exip.sql:262]: whoops

File: src/ui/server/controllers/insurance/your-buyer/connection-with-buyer/index.test.ts [hunk 5]
```diff
@@ -125,7 +125,7 @@ describe('controllers/insurance/your-buyer/connection-with-buyer', () => {
         userName: getUserNameFromSession(req.session.user),
         application: mapApplicationToFormFields(mockApplication),
         FIELD_HINT: PAGE_CONTENT_STRINGS.HINT,
-        applicationAnswer: mockApplication.buyer[CONNECTION_WITH_BUYER],
+        applicationAnswer: mockApplication.buyer.buyerRelationship[CONNECTION_WITH_BUYER],
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply src/ui/server/controllers/insurance/your-buyer/connection-with-buyer/index.test.ts:128]: its in the buyer table though

[the remark this answers is not in the document]
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/your-buyer/connection-with-buyer/index.test.ts:128]: If the GQL query is updated to query `buyerRelationship` outside of `buyer`, can simplify this

Like we do with `policyContact`

File: database/exip.sql [hunk 6]
```diff
@@ -207,24 +207,65 @@ CREATE TABLE `Buyer` (
...
+# ------------------------------------------------------------
+
+DROP TABLE IF EXISTS `BuyerContact`;
+
+CREATE TABLE `BuyerContact` (
+  `id` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
+  `contactFirstName` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `contactLastName` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `contactPosition` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `contactEmail` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
```
Comment (ttbarnes) [inline database/exip.sql:237]: Email should have 300 varchar

File: database/exip.sql [hunk 7]
```diff
@@ -207,24 +207,65 @@ CREATE TABLE `Buyer` (
...
+DROP TABLE IF EXISTS `BuyerContact`;
+
+CREATE TABLE `BuyerContact` (
+  `id` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
+  `contactFirstName` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `contactLastName` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `contactPosition` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `contactEmail` varchar(300) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+  `canContactBuyer` tinyint(1) DEFAULT NULL,
+  `application` varchar(191) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
```
Comment (ttbarnes) [inline database/exip.sql:239]: Not a blocker and sorry didn't spot this earlier, these can be renamed to simply `firstName`, `lastName` etc without the `contact` prefix 😎 

Can do in another PR

**Review by ttbarnes — APPROVED** [review]
